platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/ProblemReportingCrossProjectModelAccess.kt

```diff
@@ -163,19 +162,6 @@ class ProblemReportingCrossProjectModelAccess(
             onIsolationViolation(what)
         }
 
-        override fun equals(other: Any?): Boolean {
-            if (other === this) {
-                return true
-            }
-            if (other == null || other.javaClass != javaClass) {
-                return false
-            }
-            val project = other as ProblemReportingProject
-            return delegate == project.delegate && referrer == project.referrer // do not include `access`
```

Comment:
How important is `referrer==project.referrer`? 🤔 
This in fact breaks "transparency" of wrappers, do we still want to have it?